From 961828c68896e250046ed4d2bb90343a248c7419 Mon Sep 17 00:00:00 2001
From: Zhi Jin <zhi.jin@intel.com>
Date: Wed, 28 Nov 2012 14:25:53 +0800
Subject: audio: check the state of RTSPSource before stop it

BZ: 70058

Stopping a uninitialized RTSPSource will trigger tombstone, so
check the state of it before stopping it.

Category: aosp improvement
Domain: AUDIO_MEDIA-Framework
Origin: Internal
Upstream-Candidate: yes

Change-Id: I028f71520d1d36f8b276770f5ec5232caf43be4c
Orig-Change-Id: If0dc5cddd39c5e6a1a48e280bbb23d62bbe8778f
Signed-off-by: Zhi Jin <zhi.jin@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 .../libmediaplayerservice/nuplayer/RTSPSource.cpp  |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index 488206f..7636984 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -115,6 +115,11 @@ void NuPlayer::RTSPSource::stop() {
     if (mLooper == NULL) {
         return;
     }
+    if (mState == DISCONNECTED) {
+        ALOGI("already disconnected.");
+        return;
+    }
+
     sp<AMessage> msg = new AMessage(kWhatDisconnect, mReflector->id());
 
     sp<AMessage> dummy;
-- 
1.7.4.1

